examples/language/lib/patterns/pattern_types.dart

```diff
@@ -0,0 +1,262 @@
+// ignore_for_file: duplicate_variable_pattern, duplicate_definition, unused_local_variable, dead_code, constant_pattern_never_matches_value_type, unnecessary_null_check_pattern, pattern_never_matches_value_type, unreachable_switch_case, unused_element
+
+sealed class Shape {}
+
+class Square implements Shape {
+  final double size;
+  Square(this.size);
+}
+
+class Circle implements Shape {
+  final double size;
+  Circle(this.size);
+}
+
+class Color {
+  static const red = true;
+  static const yellow = false;
+  static const blue = true;
+}
+
+class Rect {
+  int width;
+  int height;
+  Rect({required this.width, required this.height});
+}
+
+class Point {
+  int? x;
+  int? y;
```

Comment:
Prefer defaulting to `final` for fields to catch mistakes later on.

```suggestion
  final int? x;
  final int? y;
```